PSGSuite/Public/Users/Invoke-GSUserOffboarding.ps1

```diff
@@ -0,0 +1,158 @@
+function Invoke-GSUserOffboarding {
+    <#
+    .SYNOPSIS
+    Wraps some common offboarding tasks such as random password setting, OAuth token revocation, mobile device removal, and more.
+
+    .DESCRIPTION
+    Wraps some common offboarding tasks such as random password setting, OAuth token revocation, mobile device removal, and more.
+
+    This function outputs in a log-style, timestamped format that is intended for auditability.
+
+    .PARAMETER User
+    The User to offboard
+
+    .PARAMETER Options
+    The tasks you would like to perform on the User. Defaults to the following: 'ClearASPs','ClearOAuthTokens','RemoveMobileDevices','Suspend','SetRandomPassword'
+
+    Available options:
+    * 'Full' - Performs all of the below tasks
+    * 'ClearASPs' - Clears Application Specific Passwords
+    * 'ClearOAuthTokens' - Clears OAuth tokens
+    * 'RemoveMobileDevices' - Removes Mobile Devices
+    * 'Suspend' - Suspends the user account
+    * 'SetRandomPassword' - Sets the user's account to a random password
+    * 'MoveToOrgUnit' - Moves the user to the DestinationOrgUnit specified
+    * 'SetLicense' - Sets the user to a different license
+
+    .PARAMETER DestinationOrgUnit
+    If Options include Full or MoveToOrgUnit, this is the OrgUnit that the user will be moved to.
+
+    .PARAMETER License
+    The License to set the user to.
+
+    .EXAMPLE
+    Invoke-GSUserOffboarding -User tom.fields@domain.com -Options Full -DestinationOrgUnit '/Former Employees'
+
+    Performs all of the listed tasks against user Tom Fields, including moving them to the '/Former Employees' OrgUnit and setting them to a VFE license.
+
+    .NOTES
+    Pull requests welcome for functionality enhancements!
+    #>
+    [OutputType('System.String')]
+    [CmdletBinding(SupportsShouldProcess,ConfirmImpact = "High")]
+    Param(
+        [Parameter(Mandatory,Position = 0,ValueFromPipeline,ValueFromPipelineByPropertyName)]
+        [Alias('PrimaryEmail','Mail')]
+        [string[]]
+        $User,
+        [Parameter()]
+        [ValidateSet('Full','ClearASPs','ClearOAuthTokens','RemoveMobileDevices','Suspend','SetRandomPassword','MoveToOrgUnit','SetLicense')]
+        [String[]]
+        $Options = @('ClearASPs','ClearOAuthTokens','RemoveMobileDevices','Suspend','SetRandomPassword'),
+        [Parameter()]
+        [string]
+        $DestinationOrgUnit,
+        [Parameter()]
+        [ValidateSet("G-Suite-Enterprise","Google-Apps-Unlimited","Google-Apps-For-Business","Google-Apps-For-Postini","Google-Apps-Lite","Google-Drive-storage-20GB","Google-Drive-storage-50GB","Google-Drive-storage-200GB","Google-Drive-storage-400GB","Google-Drive-storage-1TB","Google-Drive-storage-2TB","Google-Drive-storage-4TB","Google-Drive-storage-8TB","Google-Drive-storage-16TB","Google-Vault","Google-Vault-Former-Employee","1010020020")]
+        [string]
+        $License = "Google-Vault-Former-Employee"
+    )
+    Begin {
+        function New-RandomPassword {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function 'New-RandomPassword' has verb that could change system state. Therefore, the function has to support 'ShouldProcess'.](https://app.codacy.com/manual/scrthq/PSGSuite/pullRequest?prid=4742353)